### ISSUE TITLE
Unit test `TestNewClient_StopRefreshAtAnotherRequest` is not stable

### ISSUE THREAD

**Issue by denis-tingaikin** [networkservicemesh/sdk#260]
## Failure logs

```
=== FAIL: pkg/networkservice/common/refresh TestNewClient_StopRefreshAtAnotherRequest (0.19s)
    client_test.go:170: 
        	Error Trace:	client_test.go:170
        	Error:      	Condition never satisfied
        	Test:       	TestNewClient_StopRefreshAtAnotherRequest
```

## Link to CI job

https://github.com/networkservicemesh/sdk/pull/257/checks?check_run_id=719395988

**Comment by denis-tingajkin** [comment]
https://github.com/networkservicemesh/sdk/pull/259/checks?check_run_id=722428611

**Comment by denis-tingajkin** [comment]
https://github.com/networkservicemesh/sdk/pull/257/checks?check_run_id=729805883

**Comment by denis-tingajkin** [comment]
https://github.com/networkservicemesh/sdk/pull/322/checks?check_run_id=790907485

**Comment by denis-tingajkin** [comment]
Fixed by https://github.com/networkservicemesh/sdk/commit/2ab10a2ee11a1cc2a4f531a38b658c48c32b8f96

**Comment by denis-tingajkin** [comment]
https://github.com/networkservicemesh/sdk/pull/371/checks?check_run_id=854545161

**Comment by xzfc** [comment]
This issue is caused by a race condition in the method `refreshClient.Request` when the client request and a request triggered by the refresh timer happen at the same time.

Consider the following code:
```go
client := next.NewNetworkServiceClient(refresh.NewClient(ctx), nextClient)
client.Request(ctx, req1)
time.Sleep(time.Minute)
client.Request(ctx, req2)
```
With a small probability, the request would be performed as in the following time diagram:
```
time -->

req1 req1 req1 req1 req2 req1 req1 req1
^^^^ ^^^^^^^^^^^^^^ ^^^^ ^^^^^^^^^^^^^^
|    |              |    |
|    |              |    Bug: it should refresh req2 instead of req1
|    |              client.Request(ctx, req2)
|    refresh of req1
client.Request(ctx, req1)
```

It is expected to be the following:
```
req1 req1 req1 req1 req2 req2 req2 req2
```

This issue is planned to be fixed in #497.
